specs/ServerCertificate.md

```diff
@@ -0,0 +1,264 @@
+               CHECK_FAILURE(args->get_ServerCertificate(&certificate));
+
+               // Continues with the navigation to a server with a TLS certificate if
+               // the certificate is invalid and signed by badssl authority.
+               if (errorStatus == COREWEBVIEW2_WEB_ERROR_STATUS_CERTIFICATE_IS_INVALID &&
+                        (certificateAuthority.get() == L"*.badssl.com" ||
+                        certificateAuthority.get() == L"BadSSL Untrusted Root Certificate Authority"))
+               {
+                 CHECK_FAILURE(args->put_Handled(TRUE));
+               }
+               else
+               {
+                 // Cancel the request.
+                 CHECK_FAILURE(args->put_Cancel(TRUE));
+               }
+               return S_OK;
+             })
+             .Get(),
+         &m_ServerCertificateErrorToken));
+    }
+    else
+    {
+      CHECK_FAILURE(m_webview->remove_ReceivingServerCertificateError(
+        m_ServerCertificateErrorToken));
+      m_ServerCertificateErrorToken.value = 0;
+    }
+  }
+  else
+  {
+    FeatureNotAvailable();
+  }
+}
+
+// This example clears the TLS decision in response to proceeding with TLS certificate errors.
+if (m_webview)
+{.
+  CHECK_FAILURE(m_webView->ClearServerCertificateErrorOverrideCache(
+        Callback<
+            ICoreWebView2ClearServerCertificateErrorOverrideCacheCompletedHandler>(
+            [](HRESULT error, PCWSTR result) -> HRESULT {
+                CHECK_FAILURE(error);
+                if (error != S_OK)
+                {
+                  ShowFailure( error,  L"Clear server certificate error override cache failed");
+                }
+                MessageBox(nullptr, L"Cleared", L"Clear server certificate error override cache", MB_OK);
+                return S_OK;
+            })
+            .Get()));
+}
+```
+## . NET/ WinRT
+```c#
+// This example bypasses the default TLS interstitial page using the ReceivingServerCertificateError
+// event handler and continues with the navigation to a server with a TLS certificate that is signed by
+// an authority that WebView2 doesn't trust. Otherwise, cancel the request.
+private bool _isServerCertificateError = false;
+void EnableCustomServerCertificateError()
+{
+  // Safeguarding the handler when unsupported runtime is used.
+  try
+  {
+    if (!_isServerCertificateError)
+    {
+      webView.CoreWebView2.ReceivingServerCertificateError += WebView_ReceivingServerCertificateError;
+    }
+    else
+    {
+      webView.CoreWebView2.ReceivingServerCertificateError -= WebView_ReceivingServerCertificateError;
+    }
+    _isServerCertificateError = !_isServerCertificateError;
+
+    MessageBox.Show(this,
+      _isServerCertificateError ? "Custom server certificate error has been enabled" : "Custom server certificate error has been disabled",
+      "Custom server certificate error");
+  }
+  catch (NotImplementedException exception)
```

Comment:
From previous API reviews we decided to not include the `catch (NotImplementedException) ` code in C# samples since it would appear in every single one and isn't necessarily required by end dev.